src/gator/core/models/timetable.py

```diff
@@ -13,131 +13,116 @@
+        assert self.season in ['regular', 'summer']
+
+        if self.season == 'regular':
+            if self.subsession == 'first':
+                return f'{self.year}9'
+            elif self.subsession == 'second':
+                return f'{self.year}1'
+            else:
+                return f'{self.year}9-{self.year}1'
+        else:
+            if self.subsession == 'first':
+                return f'{self.year}5F'
+            elif self.subsession == 'second':
+                return f'{self.year}5S'
+            else:
+                return f'{self.year}5'
 
     def __str__(self) -> str:
-        """Return a string representation of this session.
-
-        The output string is the session code. See `code` for more
-        information.
-        """
+        """Return a string representation of the session."""
         return self.code
 
     @classmethod
-    def parse(cls, session_code: Union[str, int]) -> 'Session':
-        """Return an instance Session representing the given session code.
-
-        Raise a ValueError if the session code is not formatted properly.
+    def from_code(cls, code: str) -> 'Session':
+        """Return a session from its canonical representation.
 
         Args:
-            session_code: The session code to parse as a string or integer.
-                If the session code is an integer, it will be converted to a
-                string and zero-padded to the left if necessary. E.g. if
-                509 is passed, it will be converted to '00509' which corresponds
-                to the fall session of the year 50.
+            code: The canonical representation of the session.
 
-        Examples:
-            >>> Session.parse(509) == Session(year=50, season='fall')
-            True
-            >>> Session.parse(20201) == Session(year=2020, season='winter')
-            True
-            >>> Session.parse('20205') == Session(year=2020, season='summer')
-            True
+        Raises:
+            ValueError: If the given code is not a valid session code.
         """
-        # Ensure that the session code is a string and zero-padded to the left
-        session_code = str(session_code).zfill(5)
-
-        if len(session_code) != 5:
-            raise ValueError(
-                f'invalid session code ("{session_code}"): expected string of length 5'
-                f', not {len(session_code)}')
-        elif not session_code.isnumeric():
-            raise ValueError(
-                f'invalid session code ("{session_code}"): expected numeric string')
-        elif int(session_code[-1]) not in cls._INVERSE_SEASON_MAP:
-            raise ValueError(
-                f'invalid session code ("{session_code}"): expected code to end in '
-                f'one of {list(cls._INVERSE_SEASON_MAP.keys())}, not {session_code[-1]}')
+        if re.match(r'^\d{4}[9|1]$', code):
+            return cls(int(code[:4]), 'regular', 'first' if code[4] == '9' else 'second')
```

Comment:
We should use mappings like before to facilitate the conversion so there is one source of truth among the whole class. Same comment for the returns below.

We can define something like
```python
# Private Class Attributes:
#    _SUBSESSION_MAP: A mapping from season name to all the valid subsession codes in that season, indexed by subsession name.
#    __INVERSE_SUBSESSION_MAP: A mapping from season name to all the subsessions in that season, indexed by subsession code.
_SUBSESSION_MAP: dict[str, dict[str, str]] = {
    'regular': {'first': '9', 'second': '1'},
    'summer': {'first': '5F', 'second': '5S', 'whole': '5'}
}
_INVERSE_SUBSESSION_MAP: dict[str, dict[str, str]] = {
    s: {v: k for k, v in d.items()} for s, d in _SUBSESSION_MAP.items()
}
```
at the top of the class, and then use it like so:
```python
year, subsession_code = code[:4], code[4]
# TODO: Throw an error when subsession_code not in cls._INVERSE_SUBSESSION_MAP['regular']
return cls(int(year), 'regular', cls._INVERSE_SUBSESSION_MAP['regular'][subsession_code])
```